sycl-fusion/jit-compiler/lib/JITContext.cpp

```diff
@@ -33,3 +33,16 @@ SPIRVBinary &JITContext::emplaceSPIRVBinary(std::string Binary) {
   Binaries.emplace_back(std::move(Binary));
   return Binaries.back();
 }
+
+std::optional<SYCLKernelInfo> JITContext::getCacheEntry(CacheKeyT &Identifier) {
+  ReadLockT ReadLock{CacheMutex};
+  if (Cache.count(Identifier)) {
+    return Cache.at(Identifier);
+  }
+  return {};
+}
+
+void JITContext::addCacheEntry(CacheKeyT &Identifier, SYCLKernelInfo &Kernel) {
+  WriteLockT WriteLock{CacheMutex};
+  Cache.emplace(std::make_pair(Identifier, Kernel));
```

Comment:
```suggestion
  Cache.emplace(Identifier, Kernel);
```
Use `emplace` to build the pair